### ISSUE TITLE
cmake adds package registry with build directory instead of install directory

### ISSUE THREAD

**Issue by carandraug** [google/glog#705]
I'm building glog from source on Windows with cmake version 3.21.1.

After the initial/configure step of cmake, a package registry is installed that refers to the build directory. This key should only be added during the install target. The side effect of this is that when building other projects that depend on glog, they serach for glog on the build directory instead of the installed library. Even when `CMAKE_INSTALL_PREFIX` is manually set, the key refers to the build directory.

How to reproduce:

1. Ensure that there is no key to start with (if it does, remove it)

    $ reg query "HKCU\Software\Kitware\CMake\Packages\glog"
    ERROR: The system was unable to find the specified registry key or value.

2. Run cmake:

    $ cmake -S . -B cbuild -DCMAKE_INSTALL_PREFIX="C:\Program Files\glog"
    [...]
    -- Build files have been written to: C:/Users/vgg/Downloads/glog-0.4.0/cbuild

3. Check that the cmake package registry points to the build directory:

    $ reg query "HKCU\Software\Kitware\CMake\Packages\glog"
    
    HKEY_CURRENT_USER\Software\Kitware\CMake\Packages\glog
        0a26e0f86f27e842b3b3e99d590f7efa    REG_SZ    C:/Users/vgg/Downloads/glog-0.4.0/cbuild

What should happen is that the registry key should only be added during the install target, and it should point to the installed directory. gflags seems to do that correctly.

**Comment by sergiud** [comment]
> C:/Users/vgg/Downloads/glog-0.4.0/cbuild

In case you actually used glog 0.4.0 for your tests please try the latest version instead and report back.

> This key should only be added during the install target.

Is this your personal opinion or how did you determine this to be the desired behavior? Either way, I strongly disagree. The purpose of package registry is make projects easily discoverable. Since build directories are generally not placed in standard directories the corresponding projects need to be exported to package registry. This is also the default CMake behavior. 

The are various options that allow to disable the package registry lookup. Please consult  the CMake documentation.

> Even when `CMAKE_INSTALL_PREFIX` is manually set, the key refers to the build directory.

`CMAKE_INSTALL_PREFIX` refers to installation prefix during the install step. You actually want `CMAKE_PREFIX_PATH` which would allow bypass the package registry.

**Comment by carandraug** [comment]
> In case you actually used glog 0.4.0 for your tests please try the latest version instead and report back.

The same problem happens with glog 0.5.0.

> Is this your personal opinion or how did you determine this to be the desired behavior? Either way, I strongly disagree. The purpose of package registry is make projects easily discoverable. Since build directories are generally not placed in standard directories the corresponding projects need to be exported to package registry. This is also the default CMake behavior.

It is no longer the default in cmake since version 3.15. See [CMP0090](https://cmake.org/cmake/help/latest/policy/CMP0090.html). I'm using CMake 3.21 but because the minimum version required by glog is 3.0, and policy 0090 is not set to new, it is using the old behaviour. For reference, the way to build glog and avoid this issue is with `-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=True`

The cmake page on policy 0090 explains why this behaviour is not desirable. I've only started developing on Windows and using cmake recently so my views may be tainted by other conventions. I would give a few more reasons why I think that behaviour is not wanted:

1. In my mind, the build directory is only the place where things are built. That is a scratch place and I don't expect anything else in the system to find or use it unless I explicitly tell them. On the other hand, I would expect tools to automatically find and use things that are installed, especially if installed system-wide. If something is not installed, it should not be used, definitely not automatically. So the build directory should not be listed on the registry, or at least not by default..
2. Currently, cmake is adding glog to the registry right after the configure step. The build has not even started and may not be started at all. If an entry is to be added the registry, it should wait until the build is finished and the library is ready to use.
3. After building, I do install glog. However, the installed version does not appear on the registry. Because the registry seems to take precedence, this means that other projects find the build directory and not the installed version.

> The are various options that allow to disable the package registry lookup. Please consult the CMake documentation.

I am aware of the multiple options to control the registry and how `find_package` finds things. My point is, if glog is installed, in the absence of extra flags and options, what should be the default behaviour? Should cmake find the build directory of glog or the installed version? I think it should find the installed version.

>> Even when CMAKE_INSTALL_PREFIX is manually set, the key refers to the build directory.
>
> CMAKE_INSTALL_PREFIX refers to installation prefix during the install step. You actually want CMAKE_PREFIX_PATH which would allow bypass the package registry.

Sorry, I didn't explain it properly. I thought that if I manually specified `CMAKE_INSTALL_PREFIX` during the build of glog, it would be clearer that I meant to install it and would not add the build directory to the ackage registry. I was reporting that even if I did that, the package registry was still pointing to the build directory.
